yaml-streamly/src/Data/Yaml/Internal.hs

```diff
@@ -215,55 +209,55 @@ data ParseState = ParseState {
 
 type Parse = StateT ParseState IO
 
-requireEvent :: Event -> Parser (ReaderT JSONPath Parse) Event ()
+requireEvent :: Event -> ParserK Event Parse ()
 requireEvent e = do
-    f <- toParserK anyEvent
+    f <- ParserK.adapt anyEvent
     unless (f == Just e) $ missed (Just e)
 
 {-# INLINE anyEvent #-}
-anyEvent :: MonadCatch m => ParserD.Parser m a (Maybe a)
-anyEvent = ParserD.Parser step initial extract
+anyEvent :: MonadCatch m => Parser.Parser a m (Maybe a)
+anyEvent = Parser.Parser step initial extract
```

Comment:
This may break with parser constructor changes. Use `Parser.fromFold Fold.one` instead.